jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketConnection.java

```diff
@@ -343,21 +343,18 @@ public void run()
         fillAndParse();
     }
 
-    public void demand(long n)
+    public void demand()
     {
-        if (n <= 0)
-            throw new IllegalArgumentException("Demand must be positive");
-
         boolean fillAndParse = false;
         try (AutoLock l = lock.lock())
         {
             if (LOG.isDebugEnabled())
-                LOG.debug("demand {} d={} fp={} {} {}", n, demand, fillingAndParsing, networkBuffer, this);
+                LOG.debug("demand {} d={} fp={} {}", demand, fillingAndParsing, networkBuffer, this);
 
             if (demand < 0)
                 return;
 
-            demand = MathUtils.cappedAdd(demand, n);
+            demand = MathUtils.cappedAdd(demand, 1);
```

Comment:
I think with our other usages of no-args demand, then a second call to demand is either an ISE or ignored.    This looks like internally we are still keeping a long counter of demand?   Is there a reason for this?   Could this just be a boolean?

Reply:
Yeah I agree with all these comments.

I think maybe it would be best to get this API change merged in case we want to do a beta3 release soon, and work on the internal changes to not allow multiple demand in another PR. Unless we are delaying beta3 until next week  and then I can work on it monday.

Reply:
@lachlan-roberts Note I think beta3 will be Monday... so perhaps you can do it by then, but no need to rush the impl... so long as that is the intent.
(also the javadoc needs to describe the intended contract regardless of implementations)

Reply:
Opened Issue https://github.com/eclipse/jetty.project/issues/9965